### ISSUE TITLE
hw_tty without gpm: QUITTING without terminal being reset

### ISSUE THREAD

**Issue by vanrein** [cosmos72/twin#44]
After `QUITTING` because it found no `gpm` in `hw_tty` on the `linux` console, control is not returned to the keyboard and shell.  Nothing typed is shown, and not all platforms allow entry of commands.  It appears that the keyboard and/or display are still in "raw" mode, rather than "cooked".

**Comment by Mirppc** [comment]
You need to have gpm to have mouse support in a TTY.   There just is no option at this time.  GPM should be a dependency and you should have it installed.  As for your keyboard issue, i cant replicate as i would not even bother in tty mode without gpm.  This should be made more of a mandatory requirement for compiling.

**Comment by cosmos72** [comment]
I guess @vanrein meant "if gpm daemon is not running" when twin is compiled with gpm support and started in Linux console.
I will look at it, thanks

**Comment by vanrein** [comment]
Yes that is what I meant.  It does offer 10s to go for a session without mouse, which is workable, but if I don't go into that mode it bails out and leaves the terminal with a prompt, but otherwise locked and unworkable.

**Comment by cosmos72** [comment]
bug confirmed.
As temporary workaround, blindly typing
```
stty sane<RETURN>
```
will fix the console after twin exited due the 10 seconds timeout

**Comment by vanrein** [comment]
The console is indeed part of the game played (on us) here.  If `tty` is `/dev/console`, it complains that

```
starting display driver module `hw_tty' . . .
GPM_InitMouse() failed: unable to detect tty device
xterm_InitMouse() failed: this `linux' terminal
has no support for xterm-style mouse reporting.
```

where on `/dev/tty2` it instead reports

```
starting display driver module `hw_tty' . . .
 . . .module `hw_tty' successfully started.
```

I am bootstrapping a system on BusyBox, so there are things to fix over here.  This is after adding `/sbin/init` to the mix to get virtual terminals.  This continues to be awkward, especially because `gpm` itself is happy to be on `/dev/console` as much as on `/dev/tty2`.

As a work-around, I can at least use `tty2` for now.

**Comment by cosmos72** [comment]
Fixed in commit fc647fe0c1e196b6b60b914ab941cf67ae0b9a1c.

I opened #52 for the unrelated issue 'gpm not detected if twin started in /dev/console' reported above.
